app/presenters/ht_registration_presenter.rb

```diff
@@ -94,8 +104,15 @@ def show_detail_geoip
   end
 
   def show_detail_identity_provider
-    entity = env["HTTP_X_SHIB_IDENTITY_PROVIDER"]
-    ERB::Util.html_escape(HTInstitution.where(entityID: entity).first&.name || entity)
+    idp = env["HTTP_X_SHIB_IDENTITY_PROVIDER"] || ""
+    # Check if identity provider matches the institution picked on the signup form
+    inst = HTInstitution.where(entityID: idp).first
+    badge = ""
+    unless self.class::AUTH_EXCEPTIONS.member?(inst_id)
+      badge_id = idp == ht_institution.entityID ? :ok : :mismatch
```

Comment:
This one is definitely worth the "danger do not approve" label if they don't match